ibis/backends/duckdb/__init__.py

```diff
@@ -1074,6 +1074,56 @@ def read_sqlite(
 
         return self.table(table_name)
 
+    def read_xlsx(
+        self,
+        path: str | Path,
+        /,
+        *,
+        sheet: str | None = None,
+        range: str | None = None,
+        **kwargs,
+    ) -> ir.Table:
+        """Read an Excel file into a DuckDB table. This requires duckdb>=1.2.0.
+
+        Parameters
+        ----------
+        path
+            The path to the Excel file.
+        sheet
+            The name of the sheet to read, eg 'Sheet3'.
+        range
+            The range of cells to read, eg 'A5:Z'.
+        kwargs
+            Additional args passed to the backend's read function.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table.
+
+        See Also
+        --------
+        [DuckDB's xlsx docs](https://duckdb.org/docs/stable/guides/file_formats/excel_import)
+        """
+        path = str(path)
+        table_name = util.gen_name("read_xlsx")
+
+        kwargs = {**kwargs}
```

Comment:
This copy here is unnecessary. `kwargs` doesn't have the same problem with mutation as do other mutable arguments, because `kwargs` itself is not directly manipulable by users.